index.js

```diff
@@ -91,6 +100,28 @@ function _insertCredentials(username, userhash) {
 let cachedAgents;
 
 /**
+ * Generates and maps an internal username to the given username
+ *
+ * To protect against username collisions, this function takes
+ * a passed in username and adds a hash to it, generating an
+ * internal username.
+ *
+ * @param {string} username Username of the user.
+ *
+ * @return {string} Internal username
+ *
+ * @private
+ */
+function _generateInternalUsername(username) {
+    if (usernameMap[username]) {
+        throw new Error(`User ${username} already exists`);
+    }
+    let generatedUsername = `${username}${Date.now()}`;
```

Comment:
I think this is sufficient based on the assumption that within a test scenario no one is going to try to create the same user multiple times, and even if someone attempts to do it they'll get the error above 👍